### DISCUSSION TITLE
Translation bug in preview (snippets and smart content)

### DISCUSSION THREAD

**Discussion by wimvds** [sulu/sulu#6829] [Q&A] [answered]
There's translation bug in the preview window on Sulu v2.5+. 

When you open a page that has snippets or smart content in one language and afterwards change the language using the dropdown in the edit window, you'll see that the snippets and smart content will remain in the original language in the preview window. Here's a screenshot of the blog page on the sulu demo, first opened in **en** and then switched to **de**.

![Screenshot from 2022-09-28 09-05-44](https://user-images.githubusercontent.com/777114/192780311-bf6321b6-16a6-4bc7-9c7c-6ac89a2de495.png)

As you can see in the screenshot, the blog posts are still in English, even though the locale was switched to German.

This seems to be due to the fact that the /admin/preview/start call is only executed on the initial page load, so content will be cached for that locale, and subsequent update calls will not touch the snippets and smart content.

Not sure what the best fix is here, a quick fix seems to be to call the previewStore.start(); in JS whenever the language is toggled, but I couldn't really figure out how to do that.

Kind regards,

Wim

**Comment by alexander-schranz** [comment, marked as answer]
Thats really interesting it seems to be a problem with the `Smart Content`. The `title` and `subtitle` is correctly updated the blocks also but the smart content not. It does not even the refresh button of the preview work which is very strange. I created a screencast:

![locale-switch-preview](https://user-images.githubusercontent.com/1698337/192791335-fa035de1-9f94-40fe-9d85-568b32ecab45.gif)

Can you create an issue for it? With a short description and add the gif to it. I think the locale is maybe read in some case false but not yet sure where and how. It is strange that it works after refresh.

**Reply by wimvds** [reply]
Ok, done -> https://github.com/sulu/sulu/issues/6831.
